### PR TITLE
Add `createGitHubReader`

### PR DESCRIPTION
**Pull request by emmatown**

### CODE REVIEW

File: packages/keystatic/src/reader/generic.ts [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+    readOrThrow: async (...args) => {
+      const entry = await (read as any)(...args);
+      if (entry === null) {
+        throw new Error(
+          `Entry "${args[0]}" not found in collection "${collection}"`
+        );
+      }
+      return entry;
+    },
+    // TODO: this could drop the fs.stat call that list does for each item
```
Comment (JedWatson) [inline packages/keystatic/src/reader/generic.ts:298]: This doesn't seem relevant to the code below?

**Review by JedWatson — APPROVED** [review]

File: packages/keystatic/src/reader/generic.ts (hunk 1 above)
Comment (emmatown) [reply packages/keystatic/src/reader/generic.ts:298]: It is, `list` does checks that aren't really necessary since we immediately read the item in `all`
